tradheli- update internal RSC governor wiki (#4129)
ArduPilot/ardupilot_wiki: copter/source/docs/traditional-helicopter-internal-rsc-governor.rst

Internal RSC Governor
=====================

> ArduPilot provides an internal RSC governor (:ref:`H_RSC_MODE<H_RSC_MODE>` =4) for those applications which do not have one externally to assure constant rotor head speed. This mode requires the setup of the ``H_RSC_THRCRV_x`` and ``H_RSC_GOV_x`` parameters and the use of an :ref:`RPM sensor<common-rpm>` for the rotor head speed.  The governor maintains desired rotor speed through a proportional controller adjusted through the droop compensator and a feedforward controller that uses the throttle curve to help respond to sudden loading and unloading of the rotor system. The governor is designed to handle changes in environmental conditions and even changes in the desired RPM from the nominal settings for which the throttle curve was originally tuned.  This is accomplished through the torque compensator which adjusts the reference from the throttle curve to maintain the desired rotor speed.  The governor is designed to maintain the desired RPM within the governor range.  If the RPM falls outside this range for more than 0.5 seconds, the governor will declare the appropriate overspeed or underspeed fault and the throttle output is reverted back to the throttle curve.  The motor interlock has to be disabled to reset the governor.  

Reviewer: I dont think we talk anywhere about how to set the params actually used for throttle curve....is this set like an idleup curve , but since we have limited negative at low stick, kinda of lop-sided? probably should have a section/page and graph somewhere...I can work on that once I fully understand...

Author: The throttle curve is the same as what you would set in your transmitter for a given pitch curve. However unlike the transmitter which bases it from the position of the stick, ardupilot bases it from the limits of the collective blade pitch range given by H_COL_MIN to H_COL_MAX. So H_RSC_THRCRV_0 is the throttle output for collective at H_COL_MIN and H_RSC_THRCRV_100 is the throttle output for H_COL_MAX.

Reviewer: so for aerobatic setups, you would set the curve like a V or U (as on a non governed vehicle) and in the stab modes since the collective is limited, it would still use the correct throttle curve point for low negative pitches....got it...

> Improvements to rotor spool up were also made.  The governor uses a torque limiter to adjust the rise of torque during rotor spool up.  :ref:`H_RSC_GOV_TORQUE<H_RSC_GOV_TORQUE>` will determine how fast the rotor speed will ramp up when rotor speed reaches 50% of the rotor RPM setting. The sequence of events engaging the governor is as follows: Throttle is increased from idle according to the :ref:`H_RSC_RAMP_TIME<H_RSC_RAMP_TIME>` and will engage the clutch and start the main rotor turning. The collective should be at flat pitch and the throttle curve set to provide at least 50% of normal RPM at flat pitch. The autothrottle torque limiter will automatically activate and start accelerating the main rotor.  The spool up completes once the governor engages.  

Reviewer: ```suggestion
The governor uses a torque limiter to adjust the rise of torque during rotor spool up to assure smoothness.  :ref:`H_RSC_GOV_TORQUE<H_RSC_GOV_TORQUE>` will determine how fast the rotor speed will ramp up when rotor speed reaches 50% of the rotor RPM setting (:ref:`H_RSC_GOV_RPM<H_RSC_GOV_RPM>`). The sequence of events engaging the governor is as follows: Throttle is increased from idle according to the :ref:`H_RSC_RAMP_TIME<H_RSC_RAMP_TIME>` and the helicopter will engage its clutch and start the main rotor turning. The collective should be at flat pitch and the throttle curve set to provide at least 50% of normal RPM at flat pitch. The autothrottle torque limiter will automatically activate and start accelerating the main rotor.  The spool up completes once the governor engages.  
```

Author: done

To investigate the effect of the torque compensator, set the desired rotor speed 100-200 RPM from the rotor speed the throttle curve was tuned.  The governor torque compensator (:ref:`H_RSC_GOV_COMP<H_RSC_GOV_COMP>`) should slowly adjust the reference output to aid the droop compensator in maintaining the desired RPM.  If RPM is low or high by more than 2-5 RPM, increase this setting by 1% at a time until the governor speed matches your RPM setting. Setting the compensator too high can result in surging and throttle "hunting". Do not make large adjustments at one time.  

> The torque limiter (:ref:`H_RSC_GOV_TORQUE<H_RSC_GOV_TORQUE>`) should not need to be adjusted unless desired rotor speeds are set above the rotor speed tuned for the throttle curve. The governor failing to engage on spool up is an indication that the torque limiter is too low.  Raise this be 5% until the governor reliably engages.  It is possible to have this too high which would be indicated by immediately getting a governor overspeed fault on spool up.

Reviewer: need to explain what that fault mainfests as

Reviewer: meant how does the user know there is an overspeed fault....gcs message?

Author: Yes there is a GCS message and it defaults to the throttle curve.  I will fix that tonight.

Author: fixed
